domehunter/dome_control.py

```diff
@@ -0,0 +1,674 @@
+        return self._dome_moving
+
+###############################################################################
+# Methods
+###############################################################################
+
+    """These map directly onto the AbstractMethods created by RPC."""
+
+    def abort(self):
+        """
+        Stop everything by switching the dome motor on/off relay to off.
+
+        """
+        # TODO: consider another way to do this in case the relay fails/sticks
+        # one way might be cut power to the automationHAT so the motor relays
+        # will receive no voltage even if the relay is in the open position?
+        self._stop_moving()
+
+    def getAz(self):
+        """
+        Return current Azimuth of the Dome.
+
+        Returns
+        -------
+        float
+            The Dome azimuth in degrees.
+
+        """
+        if self.dome_az is None:
+            print("Cannot return Azimuth as Dome is not yet calibrated.\
+                   Run calibration loop")
+        return self.dome_az
+
+    def GotoAz(self, az):
+        """
+        Send Dome to a requested Azimuth position.
+
+        Parameters
+        ----------
+        az : float
+            Desired dome azimuth position in degrees.
+
+        """
+        if self.dome_az is None:
+            print('Dome is not yet calibrated, running through calibration'
+                  ' procedure, then will go to AZ specified.')
+            self.calibrate_dome_encoder_counts()
+
+        delta_az = az - self.dome_az
+        # determine whether CW or CCW gives the short path to desired az
+        if abs(delta_az) > 180:
+            if delta_az > 0:
+                delta_az -= 360
+            else:
+                delta_az += 360
+
+        # if updated delta_az is positive, direction is CW
+        if delta_az > 0:
+            # converted delta_az to equivilant in encoder ticks
+            ticks = self._az_to_ticks(delta_az)
+            target_position = self.encoder_count + ticks
+            self._move_cw()
+            # wait until encoder count matches desired delta az
+            while self.encoder_count < target_position:
+                if self.testing:
+                    # if testing simulate a tick for every cycle of while loop
+                    self._simulate_ticks(num_ticks=1)
+            self._stop_moving()
+            # compare original count to current just in case we got more ticks
+            # than we asked for
+            old_encoder_count = target_position - ticks
+            # update dome_az based on the actual number of ticks counted
+            self.dome_az += self._ticks_to_az(
+                self.encoder_count - old_encoder_count)
+            # take mod360 of dome_az to keep 0 <= dome_az < 360
+            self.dome_az %= 360
+            # update encoder_count to match the dome_az
+            self.encoder_count = self._az_to_ticks(self.dome_az)
```

Comment:
Pretty sure I just did this because I didn't want ticks to accumulate forever (like if we tend to go ccw more than we go cw or something). I can't just take mod360 of the encoder ticks so I did it this kind of goofy way.
As you can see, this is very important as if left continually running for several eons the memory allocated to the integer could exceed the total ram on the pi and cause an error.
I'll get rid of it but if that ^ happens the blood will be on your hands.